### ISSUE TITLE
Members are not listed under hardcoded group when users are imported from LDAP

### ISSUE THREAD

**Issue by FedmahnK** [keycloak/keycloak#26870]
### Before reporting an issue

- [X] I have read and understood the above terms for submitting issues, and I understand that my issue may be closed without action if I do not follow them.

### Area

import-export

### Describe the bug

Hi,

This is the same issue that this one  : https://github.com/keycloak/keycloak/issues/10348

The bug was closed in order to focus on the new store, but the bug is present in very recent version.

regards

### Version

22.0.5

### Expected behavior

- add a hardcoded-ldap-group-mapper in a ldap provider
- see the users in the define hardcoded group

### Actual behavior

The group is empty, even if it's shown in the group tab of the imported users

### How to Reproduce?

- add a hardcoded-ldap-group-mapper in a ldap provider

### Anything else?

the closed issue : https://github.com/keycloak/keycloak/issues/10348

**Comment by jonkoops** [comment]
We only support the latest version of Keycloak, please verify this issue is reproducible there.

**Comment by FedmahnK** [comment]
Hi, tested on 23.0.6, same behavior.

**Comment by jonkoops** [comment]
Thanks for confirming :+1:

**Comment by sguilhen** [comment]
I think this is similar to [this issue](https://github.com/keycloak/keycloak/issues/25005), but for groups. If you read @rmartinc 's comment here https://github.com/keycloak/keycloak/issues/25005#issuecomment-1828143842 it is said that his behavior is expected for hardcoded mappers.

@rmartinc is my assumption correct here?

**Comment by rmartinc** [comment]
@sguilhen @FedmahnK Yes, the `hardcoded-ldap-group-mapper` is like a ghost group assignment that is added to the ldap user (see the [class code](https://github.com/keycloak/keycloak/blob/23.0.7/federation/ldap/src/main/java/org/keycloak/storage/ldap/mappers/HardcodedLDAPRoleStorageMapper.java)). The user model reports the user belongs to that group (methods `isMembeOf`, `getGroupsStream`, `hasRole`,...) but it's not really assigned at database level. It is mainly used to be reported in tokens. Note that if we wanted to show those users in the group membership we would need to retrieve all users in the ldap user provider (I don't think this makes any sense). If you want real role assignment you need to really assign the user to the group.

**Comment by pedroigor** [comment]
Looks like the behavior is expected as per https://github.com/keycloak/keycloak/issues/26870#issuecomment-1959672657. The mapper does not link the group to the user but creates a wrapper around it to consider that group when checking for membership and granted roles.

Marking as an enhancement.

**Comment by nikita-kuznetsov** [comment]
I found a workaround.
Create new hardcoded-attribute-mapper, for example:
is_employee=1

Then you can get users by attribute search
/{realm}/users?q=is_employee:1

Works perfectly for me.

P.S. This workaround also fixes https://github.com/keycloak/keycloak/issues/25778 for me!!!

**Comment by gyarakilaszlo** [comment]
The same issue occurs with the `hardcoded-ldap-role-mapper`. Although the role membership is listed in the user's profile, it is not listed in the client's role. Unfortunately, the above workaround does not help me.
